test/unit/nics/managed_test.rb

```diff
@@ -101,6 +101,37 @@ class ManagedTest < ActiveSupport::TestCase
     assert_equal '11:22:33:44:55:66', n.inheriting_mac
   end
 
+  context "there is a domain" do
+    setup do
+      @domain = FactoryGirl.create(:domain)
+    end
+
+    test "host is invalid if two interfaces has same DNS name and domain" do
+      h = FactoryGirl.build(:host, :managed)
+      h.interfaces.build(:name => 'test', :domain_id => @domain.id)
+      h.interfaces.build(:name => 'test', :domain_id => @domain.id)
+      refute h.valid?
+      assert h.interfaces.any? { |i| i.errors[:name].present? }
+    end
+
+    test "host is valid if two interfaces has different DNS name and same domain" do
+      h = FactoryGirl.build(:host, :managed)
+      h.interfaces.build(:name => 'test2', :domain_id => @domain.id)
+      h.interfaces.build(:name => 'test', :domain_id => @domain.id)
+      h.valid? # trigger validation
```

Comment:
I don't think so, I can't say if the host is valid, only thing that this test setups and can test is that it does not add DNS errors